core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -77,6 +83,51 @@ public static TableMetadata newTableMetadata(
     return newTableMetadata(schema, spec, SortOrder.unsorted(), location, properties);
   }
 
+  /**
+   * Deletes the oldest metadata files if {@link
+   * TableProperties#METADATA_DELETE_AFTER_COMMIT_ENABLED} is true.
+   *
+   * @param io file IO
+   * @param base table metadata on which previous versions were based
+   * @param metadata new table metadata with updated previous versions
+   */
+  public static void deleteRemovedMetadataFiles(
```

Comment:
I don't think we should be moving this to `TableMetadata`as this adds unnecessary dependencies to `TableMetadata` (`FileIO` and `SupportsBulkOperations`). 
@leesf what's the goal of this PR and why do we want to move this method here?

Reply:
@nastra because currently the `deleteRemovedMetadataFiles` method lies both in `HadoopTableOperations` and `BaseMetastoreTableOperations` and cause duplicate code, the place i would think is moving this method here, or what do you have better idea?

Reply:
I would say maybe rather move it to `CatalogUtil`

Reply:
@nastra done